meilisearch/client.py

```diff
@@ -464,3 +468,67 @@ def wait_for_task(
             An error containing details about why Meilisearch can't process your request. Meilisearch error codes are described here: https://docs.meilisearch.com/errors/#meilisearch-errors
         """
         return wait_for_task(self.config, uid, timeout_in_ms, interval_in_ms)
+
+    def generate_tenant_token(
+        self,
+        search_rules: Dict[str, Any],
+        expires_at: Optional[int] = None,
+        api_key: Optional[str] = None
+    ) -> str:
+        """Generate a JWT token for the use of multitenancy.
+
+        Parameters
+        ----------
+        search_rules:
+            A Dictionary or an object which contains the rules to be enforced at search time for all or specific
+            accessible indexes for the signing API Key.
+            In the specific case of you want to have any restrictions you can also use a array ["*"].
+        expires_at (optional):
+            Date and time when the key will expire.
+            Note that if an expires_at value is included it should a `timestamp`.
+        api_key (optional):
+            The API key parent of the token. If you let it empty the client API Key will be used.
```

Comment:
`The API key parent of the token. If you leave it empty the client API Key will be used.`

Also, wouldn't it be safer to use the default search key instead of the API Key?

Reply:
Only the master key has access to other keys so if you use another key we don't have access to the default search key

Reply:
This means that any time a tenant token is generated where the user used the master key to create the client, and didn’t specify an `api_key` the token will fail. This will most likely be confusing to the user. Is there some way to handle this at the time of generation? It is going to be hard to know why the token isn’t working otherwise.

Reply:
I just add a note: `It's not possible to create a token from the master key.`
But I can add a check to try the keys route :
```
    try:
         client.getKeys()
```
To know if the master key is used

Reply:
We could also use the `masterKey` to get the search key and create a token but I think this is too tricky as the user will not be able to know which key the token is linked to.

Reply:
Should `api_key` be required instead of optional then? There seems to be a lot of “what if” traps by not requiring it.

Reply:
Yes, I think so. We decide with the integration team to use the client.api_key by default and add an optional `api_key`. See more [here](https://github.com/meilisearch/integration-guides/issues/181#issuecomment-1043304750).

Reply:
I don't disagree at all that this is a nice way to handle it. The problem is if the client.api_key is the master key the key will fail and the user won't know why it failed. It gets into a weird situation where sometimes the api_key is required and sometimes it isn't, but it's always listed as optional.

Reply:
After discussion with @brunoocasali, it seems better not to add this validation. Even if it could create errors in the token generation. A note will be added to the subject that a master key can not generate a token.
Moreover, if the user wants to create a key while being offline it will be possible because this function doesn't make any call to the Meilisearch API.